java/src/test/java/com/saucelabs/simplesauce/SauceOptionsTest.java

```diff
@@ -1,31 +1,92 @@
+        sauceOptions.setBrowserName(BrowserType.FIREFOX);
+        sauceOptions.setPlatformName(Platforms.MAC_OS_HIGH_SIERRA.getOsVersion());
+        sauceOptions.setBrowserVersion("68");
+
+        assertEquals(BrowserType.FIREFOX, sauceOptions.getBrowserName());
+        assertEquals("68", sauceOptions.getBrowserVersion());
+        assertEquals("macOS 10.13", sauceOptions.getPlatformName());
+    }
+
+    @Test
+    @Ignore("Not Implemented Yet")
+    public void acceptsOtherW3CValues() {
+    }
+
+    @Test
+    @Ignore("Not Implemented Yet")
+    public void acceptsSauceLabsSettings() {
+    }
 
     @Test
-    public void sauceOptions_defaultBrowser_setToChrome() {
-        assertEquals("chrome", options.getBrowserName());
+    public void acceptsSeleniumBrowserOptionsClass() {
+        FirefoxOptions firefoxOptions = new FirefoxOptions();
+        firefoxOptions.addArguments("--foo");
+        firefoxOptions.addPreference("foo", "bar");
+        firefoxOptions.setUnhandledPromptBehaviour(DISMISS);
+
+        sauceOptions = new SauceOptions(firefoxOptions);
+
+        assertEquals("firefox", sauceOptions.getBrowserName());
+        assertEquals(firefoxOptions, sauceOptions.getSeleniumCapabilities());
     }
 
     @Test
-    public void sauceOptions_defaultBrowserVersion_setToLatest() {
-        assertEquals("latest", options.getBrowserVersion());
+    public void acceptsSeleniumMutableCapabilitiesClass() {
+        MutableCapabilities caps = new MutableCapabilities();
+        caps.setCapability("browserName", "firefox");
+        SauceOptions sauceOptions = new SauceOptions(caps);
+        assertEquals(caps, sauceOptions.getSeleniumCapabilities());
+
+        caps.setCapability("browserName", "chrome");
+        assertEquals("firefox", sauceOptions.getSeleniumCapabilities().getBrowserName());
     }
 
     @Test
-    public void sauceOptions_defaultOS_setToWindows() {
-        assertEquals("Windows 10", options.getOperatingSystem());
+    @Ignore("Not Implemented Yet")
+    public void setsCapabilitiesFromMap() {
+    }
+
+    @Test
+    public void allowsBuildToBeSet() {
+        sauceOptions.setBuild("Manual Build Set");
+        assertEquals("Manual Build Set", sauceOptions.getBuild());
+    }
+
+    @Test
+    public void createsDefaultBuildName() {
+        SauceOptions sauceOptions = spy(new SauceOptions());
+        doReturn("Not Empty").when(sauceOptions).getEnvironmentVariable("BUILD_TAG");
+        doReturn("TEMP BUILD").when(sauceOptions).getEnvironmentVariable("BUILD_NAME");
+        doReturn("11").when(sauceOptions).getEnvironmentVariable("BUILD_NUMBER");
+
+        assertEquals("TEMP BUILD: 11", sauceOptions.getBuild());
     }
 
     @Test
-    public void withChrome_browser_setToChrome() {
-        options.withChrome();
-        assertNotNull(options.getChromeOptions());
+    @Ignore("Not Implemented Yet")
+    public void parsesW3CAndSauceAndSeleniumSettings() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/SauceLabs/simple_sauce/pullRequest?prid=4736477)